res/controllers/Traktor-Kontrol-S3-hid-scripts.js

```diff
@@ -15,6 +17,47 @@
 var TraktorS3 = {};
 
 // ==== Friendly User Configuration ====
+// This controller script has two modes for controlling FX:
+//
+// - A mode where the Filter and FX 1-4 buttons switch between the first five
+//   quick effect chain presets found in Settings -> Effects, and the FX Enable
+//   buttons toggle the enabled/bypass status of those quick effect chains. This
+//   emulates the intended behavior of the Mixer FX section on the Traktor
+//   Kontrol S3. The Filter button is bound to the first preset in the list, and
+//   the FX 1-4 buttons are bound to presets 2-5.
+//
+//   Pressing one of the five FX/Filter buttons switches every channel's quick
+//   effect chain to the corresponding preset. Holding one of the buttons down
+//   while pressing one of the four channel's Filter Enable buttons will assign
+//   the chain to just that channel. When the Filter Enable button is not
+//   enabled then channel will behave the same as if the first Filter quick
+//   effect chain preset was used.
+//
+//   The mode works best with the following configuration:
+//
+//   - The 'Keep superknob position' option in the Effects preferences page
+//     should be enabled.
+//   - The very first quick effect preset in the quick effect presets list
+//     should be set to the Moog Filter preset or another filter preset.
+//   - The next four quick effect presets should contain that exact same filter
+//     effect, plus another effect. Delays, reverbs, flangers, trance gates, and
+//     white noise are some examples of effects that would work well here.
+//
+// - Another mode that exposes all of Mixxx's effect controls at the expense of
+//   being more complex to use. See the manual for the full key binding scheme.
+//
+// The first mode is dubbed 'vanilla mode' as it behaves in the way the mixer FX
+// section was originally intended to be used by Native Instruments. Disable
+// this option to use the second, Mixxx-specific mode.
+TraktorS3.VanillaFxMode = true;
```

Comment:
Basic implies limited and advanced implies being a superset of the basic functionality with more options exposed. That probably gives the wrong impression. What about effect chain mode (or some wording containing quick effects)?

Reply:
So do you mean "BasicFxMode" and "EffectChainMode"?  

other ideas: "SingleFxMode" vs "MultiFxMode"; "QuickEffectsMode" vs "EffectChainMode"  or really any permutation of those haha.

Reply:
I'd probably go for quick effects mode and multi effect mode. Effect chain also doesn't seem like the right choice because the new behavior lets you toggle between effect chains while the old behavior uses fixed effect chains.